Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -300,6 +301,14 @@ public void SetOutputOptions(OutputOptions options)
 			get; private set;
 		}
 
+		public List<Assembly> Assemblies {
+			get {
+				lock (assemblies) {
+					return assemblies;
+				}
```

Comment:
returning the underlying list itself would allow the session to mutate the list in the background which would not be ideal for consumers. instead we can return an array and then simply return assemblies.ToArray() inside the lock

Reply:
@slluis   Would the ProcessInfo, in I presume currentProcesses, always be set? From what I can tell this is only set if the client calls GetProcesses()?

Reply:
Yes, currentProcesses is filled when GetProcesses() is called, but that doesn't matter. We can keep the list of assemblies internally in DebuggerSession, and in ProcessInfo just get the subset corresponding to the process, something similar to what it is being done for GetThreads().

Reply:
This should be a method, not a property, as per @slluis' initial request